### PR TITLE
applications: nrf_desktop: Restrict power modes during DFU

### PR DESCRIPTION
**Pull request by rakons**
This commit makes sure that when DFU is in progress
the CPU would not go into deep sleep when USB cable
is disconnected in the middle of the operation.

### CODE REVIEW

**Comment by wiminordic** [comment]
From test perspective this PR looks good. Issue could not be reproduced in three attempts. Pls review/merge asap.

**Review by wiminordic — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

File: applications/nrf_desktop/src/modules/dfu.c [hunk 1]
```diff
@@ -352,6 +357,10 @@ static void handle_dfu_start(const uint8_t *data, const size_t size)
 
 	sync_offset = 0;
 
+	if (IS_ENABLED(CONFIG_CAF_POWER_MANAGER_EVENTS)) {
+		power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_SUSPENDED);
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/dfu.c:361]: Should be moved after checks, I think right after `DFU started` log

**Review by pdunaj — CHANGES_REQUESTED** [review]
There is a bug here

File: applications/nrf_desktop/src/modules/dfu.c (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/dfu.c:361]: Otherwise we may prevent deep sleep forever

**Review by MarekPieta — CHANGES_REQUESTED** [review]
One fix is needed

File: applications/nrf_desktop/src/modules/dfu.c (hunk 1 above)
Comment (rakons) [reply applications/nrf_desktop/src/modules/dfu.c]: Good catch! Done.

**Review by MarekPieta — APPROVED** [review]

**Review by wiminordic — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Pull request merged**